etl/exportConsumer.go

```diff
@@ -0,0 +1,36 @@
+package etl
+
+import (
+	"fmt"
+	"sync"
+)
+
+// ExportConsumer consumes a list of transformed lines
+type ExportConsumer struct {
+	wg       sync.WaitGroup
+	incoming chan string
+}
+
+// Start causes the ExportConsumer to begin consuming payloads from the incoming channel asynchronously.
+func (ec *ExportConsumer) Start() {
+	go func() {
+		ec.wg.Add(1)
+		defer ec.wg.Done()
+
+		for payload := range ec.incoming {
+			ec.wg.Add(1)
+			go ec.consume(payload)
```

Comment:
Any reason why there's a second goroutine here? This feels like it's overcomplicating things quite a bit.

Reply:
Secondary goroutine is so that each consume task happens in its own goroutine, I want to process all filenames concurrently, all lines concurrently, all exports concurrently.